cli/cmd/common/diff.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+// Diff returns a string representation of the difference between two maps as YAML.
+// The returned string is sorted alphabetically by key. If the maps are identical, the returned string is empty.
+func Diff(a, b map[string]interface{}) (string, error) {
+	if len(a) == 0 && len(b) == 0 {
+		return "", nil
+	}
+
+	buf := new(strings.Builder)
+	err := diffRecursively(a, b, 0, buf)
+
+	return buf.String(), err
+}
+
+// diffRecursively iterates over both maps and writes the differences to the given buffer.
+func diffRecursively(a, b map[string]interface{}, recurseDepth int, buf *strings.Builder) error {
+	// Get the union of keys in a and b sorted alphabetically.
+	keys := collectKeys(a, b)
+
+	for _, key := range keys {
+		valueInA, inA := a[key]
+		valueInB, inB := b[key]
+
+		aSlice := map[string]interface{}{
+			key: valueInA,
+		}
+		bSlice := map[string]interface{}{
+			key: valueInB,
+		}
+
+		// If the key is in both a and b, compare the values.
+		if inA && inB {
+			// If the map slices are the same, write as unchanged YAML.
+			if reflect.DeepEqual(aSlice, bSlice) {
+				asYaml, err := yaml.Marshal(aSlice)
+				if err != nil {
+					return err
+				}
+
+				writeWithPrepend("  ", string(asYaml), recurseDepth, buf)
+				continue
+			}
+
+			// If the map slices are different and there is another level of depth to the map, recurse.
+			if !isMaxDepth(aSlice) && !isMaxDepth(bSlice) {
```

Comment:
Yes, but that would require reflection as far as I can tell. I might be wrong though.

Reply:
yeah you can do it with reflection (I think it'd be justified in this case). Another way is to try to cast to a map, e.g. 

```
valueInAMap, ok := valueInA.(map[string]interface{})
```
If casting is successful (`ok` is `true`), it's map. Otherwise it's not.

Reply:
Brilliant!